src/spikeinterface/core/frameslicerecording.py

```diff
@@ -34,7 +34,9 @@ def __init__(self, parent_recording, start_frame=None, end_frame=None):
         if start_frame is None:
             start_frame = 0
         else:
-            assert 0 <= start_frame < parent_size
+            assert (
+                0 <= start_frame < parent_size
+            ), f"'start_frame' must be fewer than number of samples in parent: {parent_size}"
```

Comment:
This is a bit confusing in the case where start_frame is less than 0. It should just be two different checks:
```
assert 0 <= start_frame, "Requested `start_frame` is negative (requested timestamp may be earlier than the first timestamp in the recording)"
assert start_frame < parent_size, f"`start_frame` must be fewer than number of samples in parent: {parent_size}"
```

Reply:
I thought about saying this too, but I vaguely remember we discussed (and documented) not allowing negative slicing so I wasn't sure if this point was moot or not. It can't hurt to be more explicit though.

Reply:
I get this error trying to run `recording_car.time_slice(0,1)` when the recording starts around t0=15 seconds, I'm just assuming it's giving a negative start_frame. If it's somehow failing with start_frame > parent _size, that seems like a different issue

Reply:
No I agree that they are different problems. Although your error should likely be handled in the time_slice function since saying you have negative frames/samples isn't super helpful. @h-mayorquin wrote the time_slice right? so maybe that should be dealt with at the time level instead? What do you think Alessio and Heberto? If you give a bad time you should just know it right away in my opinion rather than getting a nonsensical frame?

Reply:
Dividing the asssertion in two each one with a message revealing what is the problem is good. Having specific assertions with proper messages in the time_slice is better. I also think  all of that can be done in another PR as it is not really central to this.

Reply:
Agreed. Thanks Heberto.